Src/AutoFixture.NUnit3/AutoDataAttribute.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Builders;
+using Ploeh.AutoFixture.Kernel;
+using ITestCaseData = NUnit.Framework.Interfaces.ITestCaseData;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// This attribute uses AutoFixture to generate values for unit test parameters. 
+    /// This implementation is based on TestCaseAttribute of NUnit3
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AutoDataAttribute : NUnitAttribute, ITestBuilder, ITestCaseData, IImplyFixture
+    {
+        private readonly IFixture _fixture;
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/>
+        /// </summary>
+        public AutoDataAttribute()
+            : this(new Fixture())
+        {
+        }
+
+        /// <summary>
+        /// Construct a <see cref="AutoDataAttribute"/> with an <see cref="IFixture"/>
+        /// </summary>
+        protected AutoDataAttribute(IFixture fixture)
+        {
+            _fixture = fixture;
+            this.RunState = RunState.Runnable;
+            this.Properties = new PropertyBag();
+        }
+
+        /// <summary>
+        ///     Construct one or more TestMethods from a given MethodInfo,
+        ///     using available parameter data.
+        /// </summary>
+        /// <param name="method">The MethodInfo for which tests are to be constructed.</param>
+        /// <param name="suite">The suite to which the tests will be added.</param>
+        /// <returns>One or more TestMethods</returns>
+        public IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test suite)
+        {
+            var test = new NUnitTestCaseBuilder().BuildTestMethod(method, suite, this.GetParametersForMethod(method));
+
+            yield return test;
+        }
+
+        private TestCaseParameters GetParametersForMethod(IMethodInfo method)
+        {
+            try
+            {
+                var specimenBuilder = new SpecimenContext(this._fixture);
+                var parameterValues = method.GetParameters()
+                    .Select(p => specimenBuilder.Resolve(p.ParameterType))
```

Comment:
Wouldn't `.Select(p => specimenBuilder.Resolve(p.ParameterInfo))` be more flexible?